e2e/pvc_check.go

```diff
@@ -140,18 +173,27 @@ func pvcTest() {
 				fmt.Printf("Pod %s is running\n", pod.Name)
 
 				By("Creating a Snapshot of the block-pvc")
-				snapshot = getSampleVolumeSnapshot(pvc.Name+"-snapshot", pvc.Name, storageClass)
+				snapshotYaml := fmt.Sprintf(volumeSnapshotYAMLTemplate, "lvmblockpvc-snapshot", testNamespace, snapshotClass, "lvmblockpvc")
+				snapshot, err = getVolumeSnapshot(snapshotYaml)
 				err = crClient.Create(ctx, snapshot)
 				Expect(err).To(BeNil())
 				fmt.Printf("Snapshot %s is created\n", snapshot.Name)
 
+				By("Verifying that the Snapshot is ready")
+				Eventually(func() bool {
+					err := crClient.Get(ctx, types.NamespacedName{Name: snapshot.Name, Namespace: snapshot.Namespace}, snapshot)
+					return err == nil && *snapshot.Status.ReadyToUse
+				}, timeout, interval).Should(BeTrue())
+
 				By("Creating a clone of the block-pvc")
-				clonePvc = getSamplePvc(size, pvc.Name+"-clone", k8sv1.PersistentVolumeBlock, storageClass, "PersistentVolumeClaim", pvc.Name)
+				pvcCloneYaml := fmt.Sprintf(pvcCloneYAMLTemplate, "lvmblockpvc-clone", testNamespace, "Block", storageClass, "lvmblockpvc")
```

Comment:
should we rather create multiple template files for reach resource type rather than updating the template like this?
I think the main purpose of the go embed would be to use static assets that don't change.

Reply:
I don't think that is necessary - we can use the same template to create multiple instances of a type. 
Having a separate file for each instance could quickly get out of hand once the tests are expanded.

Reply:
ok